wgpu/tests/common/mod.rs

```diff
@@ -278,7 +278,10 @@ pub fn initialize_test(parameters: TestParameters, test_function: impl FnOnce(Te
     }
 
     let panicked = catch_unwind(AssertUnwindSafe(|| test_function(context))).is_err();
+    #[cfg(any(not(target_arch = "wasm32"), target_os = "emscripten"))]
     let canary_set = hal::VALIDATION_CANARY.get_and_reset();
+    #[cfg(not(any(not(target_arch = "wasm32"), target_os = "emscripten")))]
```

Comment:
my kingdom for an else

Reply:
In all seriousness I should check if cfg_if is in our deps.

Reply:
We did have it in our deps already, so replaced this with cfg-if